### PR TITLE
[Feature] Add Vintext Converter

### PR DESCRIPTION
**Pull request by Mountchicken**
## Motivation
Add data converter for [VinText](https://github.com/VinAIResearch/dict-guided)

## Modification
add tools for loading and converting labels from .txt annotations
```bash
tools/data/textdet/vintext_converter.py
tools/data/textrecog/vintext_converter.py
```
## Use cases (Optional)
```bash
mkdir vintext && cd vintext
# Download dataset from google drive
wget --load-cookies /tmp/cookies.txt "https://docs.google.com/uc?export=download&confirm=$(wget --quiet --save-cookies /tmp/cookies.txt --keep-session-cookies --no-check-certificate 'https://docs.google.com/uc?export=download&id=1UUQhNvzgpZy7zXBFQp0Qox-BBjunZ0ml' -O- | sed -rn 's/.*confirm=([0-9A-Za-z_]+).*/\1\n/p')&id=1UUQhNvzgpZy7zXBFQp0Qox-BBjunZ0ml" -O vintext.zip && rm -rf /tmp/cookies.txt
# Extract images and annotations
unzip -q vintext.zip && rm vintext.zip
cd vietnamese
# Rename files 
mv labels annotations && mv test_image test && mv train_images  training && mv unseen_test_images  unseen_test
mkdir imgs
mv training imgs/ && mv test imgs/ && mv unseen_test imgs/
# Delete unecessary files
rm general_dict.txt && rm vn_dictionary.txt
# Generate for detection
cd ../../..
python tools/data/textdet/vintext_converter.py data/vintext/vietnamese --nproc 4
# Generate for recognition
python tools/data/textrecog/vintext_converter.py data/vintext/vietnamese --nproc 4
```
WXWorkCapture_16468209109769

### CODE REVIEW

File: tools/data/textdet/vintext_converter.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            line = line.strip('\n')
+            ann = line.split(',')
+            bbox = ann[0:8]
+            word = line[len(','.join(bbox)) + 1:]
+            bbox = [int(_) for _ in bbox]
+            segmentation = bbox
+            x_min = min(bbox[0], bbox[2], bbox[4], bbox[6])
+            x_max = max(bbox[0], bbox[2], bbox[4], bbox[6])
+            y_min = min(bbox[1], bbox[3], bbox[5], bbox[7])
+            y_max = min(bbox[1], bbox[3], bbox[5], bbox[7])
```
Comment (xinke-wang) [inline tools/data/textdet/vintext_converter.py:124]: max not min

Comment (Mountchicken) [reply tools/data/textdet/vintext_converter.py]: My bad

File: docs/en/datasets/det.md [hunk 2]
```diff
@@ -212,3 +221,23 @@ rm dataset.zip && rm -rf dataset
...
+mv vietnamese/labels ./ && mv vietnamese/test_image ./ && mv vietnamese/train_images ./ && mv vietnamese/unseen_test_images ./
+rm -rf vietnamese
+# Rename files
+mv labels annotations && mv test_image test && mv train_images  training && mv unseen_test_images  unseen_test
+mkdir imgs
+mv training imgs/ && mv test imgs/ && mv unseen_test imgs/
+```
+- Step2: Generate `instances_test.json` and `instances_training.json`
+```bash
+python tools/data/textdet/vintext_converter.py PATH/TO/vintext --nproc 4
```
Comment (xinke-wang) [inline docs/en/datasets/det.md:242]: Generate `instances_training.json`, `instances_test.json`, and `instances_unseen_test.json`.

**Comment by xinke-wang** [comment]
please fix lint

File: tools/data/textdet/vintext_converter.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    """
+
+    with open(gt_file, 'r', encoding='utf-8') as f:
+        anno_info = []
+        for line in f:
+            line = line.strip('\n')
+            ann = line.split(',')
+            bbox = ann[0:8]
+            word = line[len(','.join(bbox)) + 1:]
+            bbox = [int(_) for _ in bbox]
```
Comment (gaotongxiao) [inline tools/data/textdet/vintext_converter.py:119]:
```suggestion
            bbox = [int(coord) for coord in bbox]
```
Don't use underscore as a referred variable name

File: tools/data/textrecog/vintext_converter.py [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+    """
+
+    with open(gt_file, 'r', encoding='utf-8') as f:
+        anno_info = []
+        for line in f:
+            line = line.strip('\n')
+            ann = line.split(',')
+            bbox = ann[0:8]
+            word = line[len(','.join(bbox)) + 1:]
+            bbox = [int(_) for _ in bbox]
```
Comment (gaotongxiao) [inline tools/data/textrecog/vintext_converter.py:121]:
```suggestion
            bbox = [int(coord) for coord in bbox]
```

**Review by xinke-wang — APPROVED** [review]

File: tools/data/textdet/vintext_converter.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+            line = line.strip('\n')
+            ann = line.split(',')
+            bbox = ann[0:8]
+            word = line[len(','.join(bbox)) + 1:]
+            bbox = [int(coord) for coord in bbox]
+            segmentation = bbox
+            x_min = min(bbox[0], bbox[2], bbox[4], bbox[6])
+            x_max = max(bbox[0], bbox[2], bbox[4], bbox[6])
+            y_min = min(bbox[1], bbox[3], bbox[5], bbox[7])
+            y_max = max(bbox[1], bbox[3], bbox[5], bbox[7])
```
Comment (gaotongxiao) [inline tools/data/textdet/vintext_converter.py:124]: They can be optionally simplified in the form of `x_min = min(bbox[0::2])`.

**Review by gaotongxiao — APPROVED** [review]

**Pull request merged**